src/main/scala/org/zalando/nakadi/client/actor/PartitionReceiverActor.scala

```diff
@@ -151,13 +156,15 @@ class PartitionReceiver (val endpoint: URI,
           bout.write(byteItem.asInstanceOf[Int])
 
           if (byteItem == '"') hasOpenString = !hasOpenString
-          else if (!hasOpenString && byteItem == '{') stack += 1
+          else if (!hasOpenString && byteItem == '{') depth += 1
           else if (!hasOpenString && byteItem == '}') {
-            stack -= 1
+            depth -= 1
 
-            if (stack == 0 && bout.size != 0) {
+            if (depth == 0 && bout.size != 0) {
               val streamEvent = objectMapper.readValue(bout.toByteArray, classOf[SimpleStreamEvent])
 
+              // Q: Can 'streamEvent.events' be null? If not, the below doesn't make sense? AKa270116
```

Comment:
yes, it can. Sometimes you do not receive the attribute at all sometimes it is just empty...I could not identify a pattern here